src/components/modebar/manage.js

```diff
@@ -73,12 +73,12 @@ function getButtonGroups(gd, buttonsToRemove, buttonsToAdd) {
     var fullLayout = gd._fullLayout,
         fullData = gd._fullData;
 
-    var hasCartesian = fullLayout._hasCartesian,
-        hasGL3D = fullLayout._hasGL3D,
-        hasGeo = fullLayout._hasGeo,
-        hasPie = fullLayout._hasPie,
-        hasGL2D = fullLayout._hasGL2D,
-        hasTernary = fullLayout._hasTernary;
+    var hasCartesian = !!fullLayout._hasCartesian,
```

Comment:
:expressionless: 

Is this out of historical necessity or are we setting some of these values to something other than booleans?